### PR TITLE
TECH-4588: add appraisal

### PR DESCRIPTION
**Pull request by ColinDKelley**
- Add Appraisal for Rails 4, 5, and 6.

### CODE REVIEW

File: .jenkins/Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    kubernetes {
+      defaultContainer "ruby"
+      yamlFile ".jenkins/ruby_build_pod.yml"
+    }
+  }
+
+  environment {
+    GITHUB_TOKEN = credentials('github_token')
+    BUNDLE_GEM__FURY__IO = credentials('gemfury_deploy_token')
+    RAILS_ENV = 'test'
```
Comment (ColinDKelley) [inline .jenkins/Jenkinsfile]: I was surprised to not find this already here. @jebentier do you know if we have it set at a higher level?

Comment (jebentier) [reply .jenkins/Jenkinsfile]: This should be set in the `spec_helper`.

File: gemfiles/rails_4.gemfile.lock [hunk 2]
```diff
@@ -0,0 +1,224 @@
+GIT
```
Comment (jebentier) [inline gemfiles/rails_4.gemfile.lock]: the `.lock` files should be ignored from github

File: git_models.gemspec [hunk 3]
```diff
@@ -21,7 +21,7 @@ Gem::Specification.new do |spec|
   spec.test_files    = spec.files.grep(%r{^(test|spec|features)/})
   spec.require_paths = ['app', 'lib']
 
-  spec.add_dependency 'rails', '>= 4.2'
-  spec.add_dependency 'hobo_fields', '~> 3.0'
-  spec.add_dependency 'git_lib', '~> 1.2'
+  spec.add_dependency 'rails',        '>= 4.2'
```
Comment (jebentier) [inline git_models.gemspec]: I've been adding `, '< 7'` to these dependencies to make sure we don't accidentally start pulling in rails 7 when it starts under developments. I think it's a good idea to keep it capped that way.

File: Appraisals [hunk 4]
```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+appraise 'rails-4' do
+  gem 'activesupport', '~> 4.2'
```
Comment (jebentier) [inline Appraisals]: should these be `rails` instead of `activesupport` since the gem depends on all of `rails`?

Comment (ColinDKelley) [reply Appraisals]: done

File: git_models.gemspec (hunk 3 above)
Comment (ColinDKelley) [reply git_models.gemspec]: done

File: gemfiles/rails_4.gemfile.lock (hunk 2 above)
Comment (ColinDKelley) [reply gemfiles/rails_4.gemfile.lock]: done

File: .jenkins/Jenkinsfile (hunk 1 above)
Comment (ColinDKelley) [reply .jenkins/Jenkinsfile]: Ok, removed from here.

**Comment by ColinDKelley** [comment]
@jebentier I addressed all your comments.

**Review by jebentier — APPROVED** [review]

**Pull request merged**